sycl/source/detail/device_filter.cpp

```diff
@@ -210,8 +212,24 @@ Parse_ONEAPI_DEVICE_SELECTOR(const std::string &envString) {
       ss << "Incomplete selector!  Try '" << Pair[0]
          << ":*' if all devices under the backend was original intention.";
       throw sycl::exception(sycl::make_error_code(errc::invalid), ss.str());
-    } else if (Pair.size() == 2) {
-      backend be = Parse_ODS_Backend(Pair[0], Entry); // Pair[0] is backend.
+    } else if (Pair.size() > 2) {
+      std::stringstream ss;
+      ss << "Error parsing selector string \"" << Entry
+         << "\"  Too many colons (:)";
+      throw sycl::exception(sycl::make_error_code(errc::invalid), ss.str());
+    }
+
```

Comment:
LGTM
NITs: 
1) why do we need extra empty line here?
2) last else if condition seems excessive - it is the only left case, we do not need a check
3) quering Pair.size() for every condition seems to be excessive too.

Reply:
1. The empty line is to separate the erroneous `Pair` sizes from the acceptable one `Pair.size() == 2`.  Just to make the code a little cleaner.
2. True - I can remove the check from the last `else` condition.
3. I expect the optimizer to optimize multiple calls to `Pair.size()` as it is immutable after construction.